configurator/sshkey/sshkey.go

```diff
@@ -0,0 +1,109 @@
+package sshkey
+
+import (
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/user"
+	"path"
+	"path/filepath"
+	"strconv"
+
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/sys/unix"
+
+	"github.com/percona/pmm-manage/configurator/config"
+)
+
+func Init(c config.PMMConfig) Handler {
+	return Handler{
+		KeyPath:  c.SSHKeyPath,
+		KeyOwner: c.SSHKeyOwner,
+	}
+}
+
+func (c *Handler) RunSSHKeyChecks() {
+	sshKeyUser, err := user.Lookup(c.KeyOwner)
+	if err != nil {
+		log.Fatal(err)
+	}
+	if c.KeyPath == "" {
+		c.KeyPath = path.Join(sshKeyUser.HomeDir, ".ssh/authorized_keys")
+	}
+
+	sshKeyDir := filepath.Dir(c.KeyPath)
+	logger := log.WithField("dir", sshKeyDir)
+	if dir, err := os.Stat(sshKeyDir); err != nil || !dir.IsDir() {
+		if err := os.MkdirAll(sshKeyDir+"/", 0700); err != nil {
+			logger.WithFields(log.Fields{
```

Comment:
fixed in https://github.com/percona/pmm-manage/pull/9/commits/8d29beadfa32a56eb0f93f2818e2ad5ffc0efbf4